src/main/java/com/spikes2212/command/drivetrains/swerve/SwerveModule.java

```diff
@@ -0,0 +1,100 @@
+package com.spikes2212.command.drivetrains.swerve;
+
+import com.spikes2212.command.DashboardedSubsystem;
+import com.spikes2212.control.FeedForwardSettings;
+import com.spikes2212.control.PIDSettings;
+import com.spikes2212.util.UnifiedControlMode;
+import com.spikes2212.util.smartmotorcontrollers.SmartMotorController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+
+import java.util.function.Supplier;
+
+public abstract class SwerveModule extends DashboardedSubsystem {
+
+    private static final double ABSOLUTE_POSITION_DISCONTINUITY_POINT = 1;
+    private static final double DEGREES_IN_ROTATIONS = 360;
+
+    protected final SmartMotorController driveMotor;
+    protected final SmartMotorController turnMotor;
+    protected final Supplier<Double> absoluteModuleAngle;
+
+    protected final boolean driveInverted;
+    protected final boolean turnInverted;
+    protected final boolean usePIDVelocity;
```

Comment:
is there a reason for this to be a class field, rather than a parameter in `setTargetVelocity()`?